pkg/scheduler/extender/extender.go

```diff
@@ -627,6 +658,40 @@ func createRequestFromPVCSpec(volumeName, storageType string, resourceRequiremen
 	}
 }
 
+func calculateScheduleTime(ctx context.Context, e *Extender, namespace string, podName string) (float64, float64, error) {
+	// record schedule information to metrics
+	cs, err := clientset.GetK8SClientset()
+	if err != nil {
+		e.logger.Errorf("cannot get clientset: %s", err)
+		return 0, 0, err
+	}
+	events, err := cs.CoreV1().Events(namespace).List(ctx, metav1.ListOptions{FieldSelector: "involvedObject.name=" + podName, TypeMeta: metav1.TypeMeta{Kind: "Pod"}})
+	if err != nil {
+		e.logger.Errorf("cannot read events for pod %s: %s", podName, err)
+		return 0, 0, err
+	}
+
+	return calculate(events.Items)
+}
+
+func calculate(events []coreV1.Event) (float64, float64, error) {
+	var totalTime, sinceLastTime float64
+
+	for _, ev := range events {
+		if ev.Reason == "Killing" {
+			// same name killed pod's events found, not count in
+			totalTime = 0
+			sinceLastTime = 0
+		} else if ev.Reason == "FailedScheduling" {
```

Comment:
For the const string,  better define it as constant.

Reply:
good advice, will add a const string to make it reusable